katdal/chunkstore_s3.py

```diff
@@ -20,7 +20,7 @@
 
 import numpy as np
 try:
-    import botocore
+    import katsdpauth.auth_botocore
```

Comment:
Did you mean `import katsdpauth.auth_botocore as botocore`? Because `botocore` still needs to come from somewhere.

Reply:
Actually seeing as you import botocore.session and botocore,config separately and you only use them in your code it works with just the katsdpauth.auth_botocore. Although if you were to try using some other function from botocore we would need to import botocore and auth_botocore. auth_botocore does import botocore, so it should still fail if the machine doesn't have botocore installed. At least with my testing so far it works.

Reply:
Good point, that initial import is just a tester - I missed the `else:` code in the review.

You still need to make `katsdpauth` optional... Something like

```Python
try:
    try:
        import katsdpauth.auth_botocore
    except ImportError:
        import botocore
    _botocore_import_error = None
except ImportError as e:
    ....
```